##############################################
# Request
##############################################

# Goal to move
string goalId

# Put it before this id, OR at this index
string beforeId
int64 toIdx

# OR: specify a complete reordering
string[] totalOrder
---
##############################################
# Response
##############################################
uint8 SUCCESS = 0
uint8 INVALID_REORDER = 1
uint8 GOAL_IN_PORGRESS = 2

uint8 returnCode